### DISCUSSION TITLE
build and flash ESP32 board

### DISCUSSION THREAD

**Discussion by dev-area** [polhenarejos/pico-hsm#45] [Q&A] [answered]
hi
i couldnt find how to build for esp32 board
also instead of entering boot mode as USB storage - it is always recognized as serial port
how to flash it?
thanks

**Comment by polhenarejos** [comment, marked as answer]
It is based on the typical process from Espressif webpage. I use VSCode plugin, which is almost "install & play".

https://docs.espressif.com/projects/esp-idf/en/v4.3/esp32/get-started/vscode-setup.html

Basically, build button, put the device in DFU mode and flash button and voila. DFU mode could be quite tricky, but my board has BOOT and RESET buttons, which enter in DFU when pressed both same time (similarly to RP2040).

If you are interested only in flashing, you can use ESP32 Flasher tool at https://www.picokeys.com/esp32-flasher/ which will flash your board without downloading any framework. Just put it in DFU and click on Flash.

**Reply by dev-area** [reply]
thanks
